src/tablestore.rs

```diff
@@ -163,10 +170,19 @@ impl TableStore {
         for chunk in encoded_sst.unconsumed_blocks {
             data.put_slice(chunk.as_ref())
         }
-        self.object_store
-            .put(&path, PutPayload::from(data))
+
+        let path = self.path(id);
+        self.transactional_wal_store
+            .put_if_not_exists(&path, Bytes::from(data))
             .await
-            .map_err(SlateDBError::ObjectStoreError)?;
+            .map_err(|e| match e {
+                object_store::Error::AlreadyExists { path: _, source: _ } => match id {
```

Comment:
Nit: Is it possible to do:

```rust
object_store::Error::AlreadyExists { .. }
```

Here?

Reply:
I tried it, I get an error without `path` and `source`.